src/main/kotlin/com/openlattice/data/DataExpiration.kt

```diff
@@ -0,0 +1,24 @@
+package com.openlattice.data
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.openlattice.client.serialization.SerializationConstants
+import com.openlattice.edm.set.ExpirationType
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+/**
+ * Holds information about the expiration parameters of data within an entity set
+ */
+
+class DataExpiration (
```

Comment:
if you make this a `data class` kotlin will automatically provide the getters/setters

Reply:
I don't think I can have the check in the class though then, right? Data classes can't have any logic in them I think.

Reply:
yep you can (ex. SimpleGraphQuery)

Reply:
wonderful thanks!